core/src/main/java/org/apache/accumulo/core/fate/FateStore.java

```diff
@@ -107,11 +118,155 @@ interface FateTxStore<T> extends ReadOnlyFateTxStore<T> {
+    // expose a FateTxStore).
+    private final UUID reservationUUID;
+    private final byte[] serialized;
+    private static final Pattern UUID_PATTERN =
+        Pattern.compile("^[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}$");
+    private static final Pattern LOCKID_PATTERN = Pattern.compile("^.+/.+\\$[0-9a-fA-F]+$");
+
+    private FateReservation(ZooUtil.LockID lockID, UUID reservationUUID) {
+      this.lockID = Objects.requireNonNull(lockID);
+      this.reservationUUID = Objects.requireNonNull(reservationUUID);
+      this.serialized = serialize(lockID, reservationUUID);
+    }
+
+    public static FateReservation from(ZooUtil.LockID lockID, UUID reservationUUID) {
+      return new FateReservation(lockID, reservationUUID);
+    }
+
+    public static FateReservation from(byte[] serialized) {
+      try (DataInputBuffer buffer = new DataInputBuffer()) {
+        buffer.reset(serialized, serialized.length);
+        ZooUtil.LockID lockID = new ZooUtil.LockID("", buffer.readUTF());
+        UUID reservationUUID = UUID.fromString(buffer.readUTF());
+        return new FateReservation(lockID, reservationUUID);
+      } catch (IOException e) {
+        throw new UncheckedIOException(e);
+      }
+    }
+
+    public static FateReservation from(String fateReservationStr) {
+      if (isFateReservation(fateReservationStr)) {
+        String[] fields = fateReservationStr.split(":");
+        ZooUtil.LockID lockId = new ZooUtil.LockID("", fields[0]);
+        UUID reservationUUID = UUID.fromString(fields[1]);
+        return new FateReservation(lockId, reservationUUID);
+      } else {
+        throw new IllegalArgumentException(
+            "Tried to create a FateReservation from an invalid string: " + fateReservationStr);
+      }
+    }
+
+    /**
+     *
+     * @param fateReservationStr the string from a call to {@link FateReservation#toString()}
+     * @return true if the string represents a valid FateReservation object, false otherwise
+     */
+    public static boolean isFateReservation(String fateReservationStr) {
+      if (fateReservationStr != null) {
+        String[] fields = fateReservationStr.split(":");
+        if (fields.length == 2) {
+          return LOCKID_PATTERN.matcher(fields[0]).matches()
+              && UUID_PATTERN.matcher(fields[1]).matches();
+        }
+      }
+      return false;
+    }
+
+    public ZooUtil.LockID getLockID() {
+      return lockID;
+    }
+
+    public UUID getReservationUUID() {
+      return reservationUUID;
+    }
+
+    public byte[] getSerialized() {
+      return serialized;
+    }
+
+    private static byte[] serialize(ZooUtil.LockID lockID, UUID reservationUUID) {
+      try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+          DataOutputStream dos = new DataOutputStream(baos)) {
+        dos.writeUTF(lockID.serialize("/"));
+        dos.writeUTF(reservationUUID.toString());
+        dos.close();
```

Comment:
Looking at existing code, this was done elsewhere (see FateKey or CompactionConfigStorage for example), and is what I based my code off of since I was doing the same thing. I also didn't understand why the close() was done early, but I left it in to be safe.

Reply:
I suspect close is called here to ensure all data is flushed in the DataOutputStream before calling `baos.toByteArray()` which is under the DataOutputStream.  Seems like calling `flush()` would be ok.

Reply:
I'll leave it as-is unless there is a significant difference between the two. If it's changed here, it should probably be changed everywhere this is done, which shouldn't be done in this PR.

Reply:
@DomGarguilo if you think it would be best to call flush everywhere can you please create and issue and resolve this comment?  There are so many comments on this issue, we are trying to get them all resolved, if no issue is needed can just resolve it.

Reply:
Since the code functions correctly and its not a big deal. I'll just resolve this without any further changes needing to be made